### PR TITLE
Make minor fixes to cache.go

### PR DESCRIPTION
**Pull request by kkaneda**
Fix errors reported by "go vet" and "errcheck". The error from "go vet" was following:

```
batchd/cache/cache.go:677: String passes lock by value: cache.SchedulerCache
```

### CODE REVIEW

File: pkg/batchd/cache/cache.go [hunk 1]
```diff
@@ -673,7 +674,7 @@ func (sc *SchedulerCache) Snapshot() *CacheSnapshot {
 	return snapshot
 }
 
-func (sc SchedulerCache) String() string {
+func (sc *SchedulerCache) String() string {
```
Comment (k82cn) [inline pkg/batchd/cache/cache.go:677]: That should not be a pointer :)

Comment (KenjiKanedaNVIDIA) [reply pkg/batchd/cache/cache.go]: Thanks for the review , @k82cn ! Hmm, if this is not a pointer receiver, why do we call `sc.Mutex.Lock` inside `String`? 

https://medium.com/golangspec/detect-locks-passed-by-value-in-go-efb4ac9a3f2b

Comment (kkaneda) [reply pkg/batchd/cache/cache.go:677]: Thanks for the review , @k82cn ! Hmm, if this is not a pointer receiver, why do we call sc.Mutex.Lock inside String?

https://medium.com/golangspec/detect-locks-passed-by-value-in-go-efb4ac9a3f2b

Comment (k82cn) [reply pkg/batchd/cache/cache.go:677]: Good point ! LGTM :)

**Review by k82cn — APPROVED** [review]

**Comment by k82cn** [comment]
@jinzhejz , do you know where do we log cache for debug?

**Comment by davidopp** [comment]
Hi @kkaneda !! Great to see you're interested in the kube-arbitrator!

**Comment by kkaneda** [comment]
Hi @davidopp ! Yes, was curious. :) Haven't yet looked very closely, but looks very promising!

**Comment by k82cn** [comment]
@kkaneda , thanks very much for the patch, I'd like to get it merged :)

**Comment by KenjiKanedaNVIDIA** [comment]
thank you, @k82cn !

**Comment by kkaneda** [comment]
thank you, @k82cn !

**Pull request merged**